analyses/focal-cn-file-preparation/04-prepare-cn-file.R

```diff
@@ -263,44 +175,45 @@ for (i in 1:length(slice_vector)){
   } else {
     end_id <- as.numeric(length(cnv_df_ids))
   }
+  
   # get the matching BS IDs
   cnv_df_ids_each <- cnv_df_ids[start_id:end_id]
+  
   # get the matching CNV dataframe
   cnv_df_each <- cnv_df %>% 
     dplyr::filter(Kids_First_Biospecimen_ID %in% cnv_df_ids_each)
   
-  # Exclude the X and Y chromosomes
+  # Exclude the X and Y chromosomes and some other contigs
   # Removing copy neutral segments saves on the RAM required to run this step
   # and file size
   cnv_no_xy_each <- cnv_df_each %>%
-    dplyr::filter(!(chr %in% c("chrX", "chrY")))
+    dplyr::filter(!(chr %in% c("chrX", "chrY")),
+                  !grepl("_", chr)) %>%
+    distinct()
   
   # Merge and annotated no X&Y
-  autosome_annotated_cn_each <- process_annotate_overlaps(cnv_df = cnv_no_xy_each,
-                                                     txdb_exons = tx_exons) %>%
+  autosome_annotated_cn_each <- process_annotate_overlaps(cnv_df = cnv_no_xy_each, exon_granges = gencode_gtf) %>%
     # mark possible amplifications in autosomes
     dplyr::mutate(status = dplyr::case_when(
       copy_number > (2 * ploidy) ~ "amplification",
       copy_number == 0 ~ "deep deletion",
-      TRUE ~ as.character(status)
-    ))
+      TRUE ~ as.character(status)))
   autosome_annotated_cn <- bind_rows(autosome_annotated_cn, autosome_annotated_cn_each)
 }
 
 # Output file name
-if (opt$runWXSonly){
-  autosome_output_file <- paste0(opt$filename_lead, "_wxs_autosomes.tsv.gz")
-}else{
-  autosome_output_file <- paste0(opt$filename_lead, "_autosomes.tsv.gz")
+if(runWXSonly){
+  autosome_output_file <- paste0(filename_lead, "_wxs_autosomes.tsv.gz")
+} else {
+  autosome_output_file <- paste0(filename_lead, "_autosomes.tsv.gz")
 }
 
 # Save final data.frame to a tsv file
-readr::write_tsv(autosome_annotated_cn,
-                 file.path(results_dir, autosome_output_file))
+readr::write_tsv(x = autosome_annotated_cn, file = file.path(results_dir, autosome_output_file))
```

Comment:
```suggestion
readr::write_tsv(autosome_annotated_cn, file.path(results_dir, autosome_output_file))
```
when running on the Docker, it throws an error message 
```
Error in readr::write_tsv(x = autosome_annotated_cn, file = file.path(results_dir,  : 
  unused argument (file = file.path(results_dir, autosome_output_file))
``` 
- modifying to original format works for me.

Reply:
That is super weird, I am pretty sure x and file are valid parameters for write_tsv. I guess the Dockerfile must be using a very old version of readr.

Reply:
I just committed this change - thanks. Like you said, it wasn't working on the docker image.